tests/test_opal_api.py

```diff
@@ -0,0 +1,83 @@
+import json
+
+import pandas as pd
+import pytest
+from fastapi.testclient import TestClient
+
+from datahub import data as dt
+from datahub.main import app
+from datahub.opal import create_opal_frame
+
+client = TestClient(app)
+client.headers["Content-Type"] = "application/json"
+
+
+@pytest.fixture(autouse=True)
+def reset_opal_data():
+    """Pytest Fixture for resetting Opal data global variable."""
+    dt.opal_df = create_opal_frame()
+
+
+def test_post_opal_api(opal_data):
+    """Tests POSTing Opal data to API."""
+    post_data = json.dumps(opal_data.copy())
+
+    """Checks that the Opal global variable can be accessed."""
```

Comment:
By convention, we tend to use `#` for in-line comments and `"""` are reserved for docstrings.
```suggestion
    # Checks that the Opal global variable can be accessed.
```